### ISSUE TITLE
Markdown UTF-8

### ISSUE THREAD

**Issue by muv1** [michaelrsweet/htmldoc#383]
If I have a Markdown file like this:

~~~
What’s the use?
~~~

Run this command:

~~~
htmldoc a.md
~~~

Result:

~~~html
<p>What&acirc;��s the use?</p>
~~~

Run this command:

~~~
htmldoc --charset utf-8 a.md
~~~

Result:

~~~html
<p>What�s the use?</p>
~~~

**Comment by michaelrsweet** [comment]
@muv1 What version of HTMLDOC are you using?

What operating system?

Can you attach the markdown file (not sure that Github is preserving the exact UTF-8 you are supplying)?

**Comment by muv1** [comment]
> What version of HTMLDOC are you using?
>
> What operating system?

~~~
htmldoc-1.9.8-windows.msi
~~~

> Can you attach the markdown file

<https://github.com/michaelrsweet/htmldoc/files/4472822/a.zip>

**Comment by michaelrsweet** [comment]
Hmm, doesn't seem to matter whether this character is in a Markdown or HTML file. Investigating...

**Comment by michaelrsweet** [comment]
OK, the sequence is E2 80 99 which translates to U+2019, which *should* be OK (translates to the "quoteright" character glyph).  And I forgot to use the --charset option so it doesn't happen for HTML, just for Markdown...

**Comment by michaelrsweet** [comment]
OK, found the bug - the markdown code that mapped UTF-8 to the internal (dynamic) character set had a bug for 3-byte UTF-8 sequences.

**Comment by michaelrsweet** [comment]
[master cbfb938] Fix Markdown UTF-8 bug (Issue #383)

**Comment by muv1** [comment]
Thanks for the fix. I ended up going with another tool for my needs:

<https://github.com/dompdf/dompdf>
